vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt

```diff
@@ -136,20 +137,20 @@ class DecryptionFailureTracker @Inject constructor(
                     // for now we ignore events already reported even if displayed again?
                     .filter { alreadyReported.contains(it).not() }
                     .forEach { failedEventId ->
-                        analyticsTracker.capture(Error(failedEventId, Error.Domain.E2EE, aggregation.key))
+                        analyticsTracker.capture(Error(aggregation.key.first, Error.Domain.E2EE, aggregation.key.second))
                         alreadyReported.add(failedEventId)
                     }
         }
     }
 
-    private fun MXCryptoError.ErrorType.toAnalyticsErrorName(): Error.Name {
+    private fun MXCryptoError.ErrorType.toAnalyticsErrorName(): DetailedErrorName {
+        val detailed = "$name | mxc_crypto_error_type"
         return when (this) {
-            MXCryptoError.ErrorType.UNKNOWN_INBOUND_SESSION_ID -> Error.Name.OlmKeysNotSentError
-            MXCryptoError.ErrorType.OLM                        -> {
-                Error.Name.OlmUnspecifiedError
-            }
-            MXCryptoError.ErrorType.UNKNOWN_MESSAGE_INDEX      -> Error.Name.OlmIndexError
-            else                                               -> Error.Name.UnknownError
+            MXCryptoError.ErrorType.UNKNOWN_INBOUND_SESSION_ID,
+            MXCryptoError.ErrorType.KEYS_WITHHELD         -> Pair(detailed, Error.Name.OlmKeysNotSentError)
+            MXCryptoError.ErrorType.OLM                   -> Pair(detailed, Error.Name.OlmUnspecifiedError)
+            MXCryptoError.ErrorType.UNKNOWN_MESSAGE_INDEX -> Pair(detailed, Error.Name.OlmIndexError)
+            else                                          -> Pair(detailed, Error.Name.UnknownError)
```

Comment:
Just about the syntax, I think you can keep and store the previous code into a val `errorName` and at the end return the pair `detailed to errorName`. Otherwise, as you defined a typealias, you can probably use it instead of calling the Pair?

Reply:
Thanks, I refactored that part